.travis.yml

```diff
@@ -0,0 +1,42 @@
+sudo: false
+
+addons:
+  apt:
+    packages:
+      - oracle-java8-installer
+
+language: java
+
+dist: trusty
+
+jdk:
+  - openjdk8
+
+install: true # skips travis' default installation step which executes gradle assemble.
+
+script: ./gradlew clean build
+
+# The before_cache and the cache steps cache the gradle installation on travis.
+before_cache:
```

Comment:
Why? According to Travis, this is correct position for `before_cache`. It's executed after `script`: https://docs.travis-ci.com/user/customizing-the-build/#The-Build-Lifecycle